addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationEnvironment.java

```diff
@@ -197,6 +216,27 @@ public ContextWrapper parseContextData(
         return wrapper;
     }
 
+    private ArrayList<?> verifyRegexes(Object value, String key, AutomationProgress progress) {
+        if (!(value instanceof ArrayList)) {
+            progress.error(Constant.messages.getString("automation.error.context." + key, value));
+        } else {
+            ArrayList<?> regexes = (ArrayList<?>) value;
+            for (Object regex : regexes) {
+                try {
+                    Pattern.compile(regex.toString());
+                } catch (PatternSyntaxException e) {
+                    progress.error(
+                            Constant.messages.getString(
+                                    "automation.error.context.badregex",
+                                    regex.toString(),
+                                    e.getMessage()));
+                }
+            }
+            return regexes;
+        }
+        return null;
```

Comment:
This could be moved to the if-then statement and the else removed (avoids the need to scan the whole method to know the return value and reduces unnecessary nesting).